packages/react-server/src/ReactFlightServer.js

```diff
@@ -4264,57 +4291,77 @@ function emitTimeOriginChunk(request: Request, timeOrigin: number): void {
 
 function forwardDebugInfo(
   request: Request,
-  id: number,
+  task: Task,
   debugInfo: ReactDebugInfo,
 ) {
+  const id = task.id;
+  const minimumTime =
+    enableProfilerTimer && enableComponentPerformanceTrack ? task.time : 0;
   for (let i = 0; i < debugInfo.length; i++) {
     const info = debugInfo[i];
     if (typeof info.time === 'number') {
       // When forwarding time we need to ensure to convert it to the time space of the payload.
-      emitTimingChunk(request, id, info.time);
+      // We clamp the time to the starting render of the current component. It's as if it took
+      // no time to render and await if we reuse cached content.
+      emitTimingChunk(
+        request,
+        id,
+        info.time < minimumTime ? minimumTime : info.time,
+      );
     } else {
-      request.pendingChunks++;
```

Comment:
@unstubbable The bug was that this was now mismatched since one of the branches doesn't emit a chunk. We should really move all these into the `emit` functions.

Reply:
Yeah, it's not the first time that managing this counter has caused a bug.